### DISCUSSION TITLE
How to implement  method interception, like spring aop

### DISCUSSION THREAD

**Discussion by zhangtianxiao** [activej/activej#168] [Q&A]

**Comment by eduard-vasinskyi** [comment]
Hello, @zhangtianxiao

The usage of AOP is completely orthogonal to ActiveJ. You can use any AOP framework of choice in conjunction with ActiveJ.

As for method interception, ActiveJ follows an OOP approach. You can use transformers (wrappers) around your classes to change a method's behavior. This is achieved via binding transformers in ActiveJ Inject. You may transform any dependency to some wrapper/decorator that would change the behavior of some instance methods.

We have added [an example](https://github.com/activej/activej/blob/v4.3-examples/examples/core/inject-2/src/main/java/BindingTransformationExample.java) that demonstrates how you can transform a dependency using DI and override instance methods to add additional logging (for example). This example is for ActiveJ v4.3, note that we have significantly simplified DI transformers for ActiveJ v5.0 release, at the same time making it more powerful.
